### ISSUE TITLE
Tooltips should be (reliably and deliberately) announced to screen reader users

### ISSUE THREAD

**Issue by Malvoz** [Leaflet/Leaflet#8112]
**Is your feature request related to a problem? Please describe.**

When tooltips are toggled visible they're not reliably announced by screen readers.

This is because screen readers generally don't announce elements on DOM updates or changes to CSS (e.g. `display`).
(I.e. https://github.com/Leaflet/Leaflet/issues/8111 doesn't help the case for screen reader users.)

(I say "not _reliably_ announced" because a side effect of the focus loss described in https://github.com/Leaflet/Leaflet/issues/8114 causes Narrator to announce _all_ tooltips when only 1 is toggled opened. OTOH, in NVDA tooltips are not announced at all after being toggled visible.)

**Describe the solution you'd like**

For each open tooltip:

1. Set [`role="tooltip"`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tooltip_role) (on `leaflet-tooltip`)
2. Set [`aria-describedby`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-describedby) on the bound element(s) with an `id` reference to the tooltip(s)
  (space-separated `id` references if pointing to multiple tooltips)

<details>
<summary>Caveat: <i>standalone</i> tooltips</summary><br>

This proposed solution is not applicable to standalone tooltips since there's no element to be described using `aria-describedby`. No natural element anyway, could perhaps be hacked around... Anyway, this is the safest option, and as such another reason for implementers to [_try to avoid standalone tooltips_](https://github.com/Leaflet/Leaflet/discussions/8006#:~:text=Mention%20that%20standalone%20popups%20and%20tooltips%20can%20be%20problematic).

<hr>
</details>

**Describe alternatives you've considered**

<details>
<summary>1. Make the <code>leaflet-tooltip-pane</code> an <a href="https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions">ARIA Live region</a></summary><br>

By setting it to [`role="status"`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_status_role), and [`aria-live="polite"`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-live) per [MDN's compatibility notes](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions#roles_with_implicit_live_region_attributes):
   
   > To maximize compatibility, add a redundant `aria-live="polite"` when using this role.

   This will result in tooltips being announced to screen reader users when they're toggled visible.

   However, this solution could potentially have negative effects on screen reader users if the Leaflet implementer is toggling tooltips irresponsibly (e.g. toggling without user interactions) or other unexpected ways. Additionally, this solution expects tooltips to exist in the default tooltip pane.

<hr>
</details>

<details>
<summary>2. Focus delegation</summary><br>
   
   I don't believe this is a good option, tooltips should NOT receive focus, at least this is true for browsers' native tooltips, and per the ARIA [best practices for tooltips](https://www.w3.org/TR/wai-aria-practices/#tooltip):
   
   > Tooltip widgets do not receive focus.

<hr>
</details>

## Additional context

Accessible tooltips using `role="tooltip"` and `aria-describedby`: https://zoebijl.github.io/apg-tooltip/.
